scripts/prepare-components.js

```diff
@@ -0,0 +1,206 @@
+const glob = require('glob');
+const path = require('path');
+const fs = require('fs');
+
+// Load copy of package.json
+let packageJsonFile = fs.readFileSync(
+	path.join(__dirname, '../package.json'),
+	'utf8'
+);
+if (packageJsonFile) {
+	packageJsonFile = JSON.parse(packageJsonFile);
+	packageJsonFile.components = [];
+}
+
+// Get all component configuration files
+const configFiles = glob.sync(
+	`${path.join(__dirname, '../components')}/**/component.json`
+);
+configFiles.forEach((filePath, index) => {
+	const data = fs.readFileSync(filePath, 'utf8');
+	if (data) {
+		configFiles[index] = JSON.parse(data.toString());
+	} else {
+		configFiles[index] = {};
+	}
+});
+
+// Read index and storybook-stories files
+let indexFile = fs.readFileSync(
+	path.join(__dirname, '../components/index.js'),
+	'utf8'
+);
+let storybookStoriesFile = fs.readFileSync(
+	path.join(__dirname, '../components/storybook-stories.js'),
+	'utf8'
+);
+
+// Create initial site-stories string
+let siteStoriesFile = `// This object is imported into the documentation site. An example for the documentation site should be part of the pull request for the component. The object key is the component value listed in the component's component.json file. The following uses webpack's raw-loader plugin to get "text files" that will be eval()'d by CodeMirror within the documentation site on page load.
+
+/* eslint-env node */
+/* eslint-disable global-require */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable import/no-unresolved */
+
+const documentationSiteLiveExamples = {
+`;
+
+// Returns a regular expression that can be used to search for specific exports in files
+const getExportRegExp = (line) => {
+	let formattedLine = line.replace(/\//g, '\\/');
+	formattedLine = formattedLine.replace(/\./g, '\\.');
+	return new RegExp(`${formattedLine}\r?\n?`);
+};
+
+// Make sure components are listed in index.js
+const checkIndex = indexFile
```

Comment:
This would be less brittle if you used `babel` to examine the parsed AST of the file itself instead of using regular expressions on the file text as a string.  You can see an example of how that works in this project in this file: https://github.com/salesforce/design-system-react/blob/master/scripts/code/find-and-fix-missing-doc-stories.js#L138 . It would make this code shorter too.

Reply:
Converted to use Babel AST 😄